x-pack/plugins/transform/public/alerting/transform_health_rule_type/transform_health_rule_trigger.tsx

```diff
@@ -0,0 +1,131 @@
+import { GetTransformsResponseSchema } from '../../../common/api_schemas/transforms';
+import { i18n } from '../../../../../../../../../../private/var/tmp/_bazel_darnautov/1afe62330ff0d9ae1ca2013aad33fd76/execroot/kibana/bazel-out/darwin-fastbuild/bin/packages/kbn-i18n';
+import { ALL_TRANSFORMS_SELECTION } from '../../../common/constants';
+
+export type TransformHealthRuleTriggerProps = AlertTypeParamsExpressionProps<TransformHealthRuleParams>;
+
+const TransformHealthRuleTrigger: FC<TransformHealthRuleTriggerProps> = ({
+  alertParams,
+  setAlertParams,
+  errors,
+}) => {
+  const formErrors = Object.values(errors).flat();
+  const isFormInvalid = formErrors.length > 0;
+
+  const api = useApi();
+  const toast = useToastNotifications();
+  const [transformOptions, setTransformOptions] = useState<string[]>([]);
+
+  const onAlertParamChange = useCallback(
+    <T extends keyof TransformHealthRuleParams>(param: T) => (
+      update: TransformHealthRuleParams[T]
+    ) => {
+      setAlertParams(param, update);
+    },
+    [setAlertParams]
+  );
+
+  useEffect(
+    function fetchTransforms() {
+      let unmounted = false;
+      api
+        .getTransforms()
+        .then((r) => {
+          if (!unmounted) {
+            setTransformOptions(
+              (r as GetTransformsResponseSchema).transforms.filter((v) => v.sync).map((v) => v.id)
+            );
+          }
+        })
+        .catch((e) => {
+          toast.addError(e, {
+            title: i18n.translate(
+              'xpack.transform.alertingRuleTypes.transformHealth.fetchErrorMessage',
+              {
+                defaultMessage: 'Unable to fetch transforms',
+              }
+            ),
+          });
+        });
+      return () => {
+        unmounted = true;
+      };
+    },
+    [api, toast]
+  );
+
+  const excludeTransformOptions = useMemo(() => {
+    if (alertParams.includeTransforms?.some((v) => v === ALL_TRANSFORMS_SELECTION)) {
+      return transformOptions;
+    }
+    return null;
+  }, [transformOptions, alertParams.includeTransforms]);
+
+  return (
+    <EuiForm
+      data-test-subj={'transformHealthAlertingRuleForm'}
+      invalidCallout={'none'}
+      error={formErrors}
+      isInvalid={isFormInvalid}
+    >
+      <TransformSelectorControl
+        label={
+          <FormattedMessage
+            id="xpack.transform.alertTypes.transformHealth.testsSelection.enableTestLabel"
+            defaultMessage="Include transforms"
+          />
+        }
+        options={transformOptions}
+        selectedOptions={alertParams.includeTransforms ?? []}
+        onChange={onAlertParamChange('includeTransforms')}
+        allowSelectAll
+      />
+
+      <EuiSpacer size="m" />
+
+      {!!excludeTransformOptions?.length || !!alertParams.excludeTransforms?.length ? (
+        <>
+          <TransformSelectorControl
+            label={
+              <FormattedMessage
+                id="xpack.transform.transformSelector.formControlLabel"
```

Comment:
This needs a better ID - something like `xpack.transform.alertTypes.transformHealth.excludeTransformsLabel`.

Reply:
Fixed in https://github.com/elastic/kibana/pull/112277/commits/6c27a241a1860d001eb6bcd84fec2cb8ec504b0c